Networking/Networking/Model/Product/Product.swift

```diff
@@ -266,17 +266,18 @@ public struct Product: Codable {
 
         let regularPrice = try container.decodeIfPresent(String.self, forKey: .regularPrice)
 
+        let onSale = try container.decode(Bool.self, forKey: .onSale)
+
         // Even though a plain install of WooCommerce Core provides string values,
         // some plugins alter the field value from String to Int or Decimal.
         var salePrice = ""
         if let parsedSalePriceString = container.failsafeDecodeIfPresent(stringForKey: .salePrice) {
-            salePrice = parsedSalePriceString
+            salePrice = (onSale && parsedSalePriceString.isEmpty) ? "0" : parsedSalePriceString
         } else if let parsedSalePriceDecimal = container.failsafeDecodeIfPresent(decimalForKey: .salePrice) {
-            salePrice = NSDecimalNumber(decimal: parsedSalePriceDecimal).stringValue
+            let stringPrice = NSDecimalNumber(decimal: parsedSalePriceDecimal).stringValue
+            salePrice = onSale ? "0" : stringPrice
```

Comment:
Sorry, what I meant was that we can revert all the previous change because the line with `NSDecimalNumber` is enough to properly parse the `sale_price`. The `onSale` condition in here also causes a bug in this scenario:

1. Change `sale_price` in the unit test JSON like so:

    ```json
            "sale_price": 12.01,
    ```
2. Run the `testThatProductSalePriceIsProperlyParsed()`

The test still passes but it shouldn't because the `product.salePrice` should be `"12.01"`. 

I believe this is enough:

```suggestion
            salePrice = NSDecimalNumber(decimal: parsedSalePriceDecimal).stringValue
```